woocommerce/woocommerce-ios
# Google Ads: Add tracking

Closes: #13273 


## Description

This PR adds tracking events for the Google Ads integration in the web view.

Event | Trigger | Properties
-- | -- | --
googleads_entry_point_displayed | When the Google for Woo entry point is shown to the user. | source
googleads_entry_point_tapped | When the Google for Woo entry point is tapped by the user and opens to campaign creation. | source, type, has_campaigns
googleads_flow_started | When the Google for Woo webview is first loaded. | source
googleads_flow_canceled | When the Google for Woo webview is dismissed without completing the flow. | source
googleads_campaign_creation_success | When the Google for Woo webview is automatically dismissed after completing the creation | source
googleads_flow_error | When the Google for Woo webview returns an error. | source

To support tracking web flow errors, `DefaultAuthenticatedWebViewModel` and `WebViewHostingController` has been updated with a new closure to be triggered when loading a web page fails.

## Testing steps

- Pre-requisite: Install, activate, and set up the Google Ads plugin to a test site following pe5sF9-2Qo-p2. Ensure that you're using the plugin version 2.7.5 or above.
- Build the app and log in to your store.
- Enable Google Ads card on the dashboard if you haven't already.
- Confirm that `googleads_entry_point_displayed` is tracked with `source: my_store`.
- Tap on either the Create campaign button or existing campaign detail, and confirm that `googleads_entry_point_tapped` is tracked with `source: my_store` and the correct value for `has_campaigns` depending on whether your store has existing campaigns.
- Switch to the menu tab, confirm that `googleads_entry_point_displayed` is tracked with `source: more_menu`.
- Tap on the Google for WooCommerce item, confirm that `googleads_entry_point_tapped` is tracked with `source: my_store` and the correct value for `has_campaigns` depending on whether your store has existing campaigns.
- When the web view first loaded, confirm that `googleads_flow_started` is tracked with the correct `source` property.
- Dismiss the web view, confirm that `googleads_flow_canceled` is tracked with the correct `source` property.
- Create a campaign on the web view. After the creation completes, confirm that `googleads_campaign_creation_success` is tracked with the correct `source` property.
- Turn off the internet connection and tap on the entry point again. After the loading fails, confirm that `googleads_flow_error` is tracked with the correct `source` property and error details.

## Screenshots

N/A

---
